ArkEcosystem/explorer
# feature: implement coingecko api

## Summary

https://app.clickup.com/t/11ty6n8

- Refactor the current coinmarketcap implementation to use a bindable contract so we can switch between apis if necessary
- Implements coingecko
- Adds a new `.env` variable for the case we want to change the default service (that is coingecko for the moment) (for example: `EXPLORER_CRYPTO_MARKET_API_SERVICE="\\App\\Services\\MarketDataServices\\CryptoCompare"`)


### To test:
1. fetch this branch
2. run the queue `php artisan horizon`
3. rin this command `php artisan cache:clear && php artisan explorer:cache-prices && php artisan explorer:cache-currencies-data && php artisan explorer:cache-currencies-history --no-delay` (it will load all the prices from coingecko)
4. Ensure the values that we got make sense for every currency
5. If you want you can also test if you can move to CryptoCompare by setting `EXPLORER_CRYPTO_MARKET_API_SERVICE="\\App\\Services\\MarketDataServices\\CryptoCompare"` on the .env file and running the steps above again


## Checklist
